app/routes.py

```diff
@@ -1,2 +1,61 @@
-from flask import Blueprint
+from app import db
+from app.models.planet import Planet
+from flask import Blueprint, jsonify, abort, make_response, request
 
+planet_bp = Blueprint("planets", __name__, url_prefix= "/planets")
+
+@planet_bp.route("", methods=["GET"])
+def get_all_planets():
+    all_planets = []
+    if len(request.args) != 0:
+        planets = Planet.query.filter_by(**request.args)
+    else:
+        planets = Planet.query.all()
+
+    for planet in planets:
+        all_planets.append(planet.to_json())
+
+    return jsonify(all_planets)
+
+@planet_bp.route("/<planet_id>", methods=["GET"])
+def get_one_planet(planet_id):
+    one_planet = Planet.validate_id(planet_id)
+
+    return one_planet.to_json()
+
+@planet_bp.route("", methods=["POST"])
+def create_planet():
+    planet_data = request.get_json()
+    new_planet = Planet(
+        name=planet_data['name'],
+        description=planet_data['description'],
+        has_life=planet_data['has_life']
+        )
```

Comment:
You can pull this into a class method in Planet, name it something like create() - here's an example of how that looks for the [Cat model in Flasky](https://github.com/ada-c17/flasky/commit/6d43a9fe543bab8d5a8ec94484267899be483063#diff-7d18e19033b5c348d84d50ef5c7e523b123c34305df76fb7eff28b7fafe09beeR28-R36)

Also, for Task List, you might consider how you could add in validation to make sure that all the required fields are sent in the request to your server.

For example, if someone sent a POST request but left off has_life, then line 32 would throw KeyError that it couldn't find has_life.

it would be nice to handle the error and return a message so that the client knows their request was invalid and they need to include has_life.